src/backend/EdFi.DmsConfigurationService.Backend.Keycloak/ClientRepository.cs

```
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using Keycloak.Net.Models.Clients;
+using Keycloak.Net.Models.Roles;
+using Keycloak.Net;
+
+namespace EdFi.DmsConfigurationService.Backend.Keycloak;
+
+public class ClientRepository(KeycloakContext keycloakContext) : IClientRepository
+{
+    private readonly KeycloakClient _keycloakClient = new(keycloakContext.Url, keycloakContext.ClientSecret,
+       new KeycloakOptions(adminClientId: keycloakContext.ClientId));
+    private readonly string _realm = keycloakContext.Realm!;
+
+    public async Task<bool> CreateClientAsync(string clientId, string clientSecret, string displayName)
+    {
+        try
+        {
+            var realmRoles = await _keycloakClient.GetRolesAsync(_realm);
+
+            var client = new Client
+            {
+                ClientId = clientId,
+                Enabled = true,
+                Secret = clientSecret,
+                Name = displayName,
+                ServiceAccountsEnabled = true,
+                ProtocolMappers = ConfigServiceProtocolMapper()
+            };
+
+            Role? clientRole = realmRoles.FirstOrDefault(x => x.Name.Equals(keycloakContext.ServiceRole, StringComparison.InvariantCultureIgnoreCase));
```

What happens if `clientRole` is `null`? Should we throw an exception?